src/components/shared/LessonSide/LevelSelector.tsx

```diff
@@ -1,5 +1,42 @@
-function LevelSelector(): JSX.Element {
-  return <div>Level Selector</div>;
+import '../../../styles/LessonSide.scss';
+
+interface LevelSelectorProps {
+  numLevels: number;
+  currentLevel: number;
+}
+
+function LevelSelector({
+  numLevels,
+  currentLevel,
+}: LevelSelectorProps): JSX.Element {
+  return (
+    <div className="selector-box">
+      <div className="level-selector">
+        {Array(numLevels * 2 - 1)
+          .fill(0)
+          .map((_, i) => {
+            const level = i / 2 + 1;
+            return i % 2 == 1 ? (
+              <div
+                className={
+                  'level-connector' +
+                  (level < currentLevel ? ' reached-level-connector' : '')
+                }
+              ></div>
+            ) : (
+              <div
```

Comment:
Can we add a comment regarding what this conditional statement switches between?